src/main/scala/viper/silver/parser/Resolver.scala

```diff
@@ -804,6 +785,8 @@ case class TypeChecker(names: NameAnalyser) {
         pq._typeSubstitutions = pq.body.typeSubstitutions.toList.distinct
         pq.typ = Bool
         curMember = oldCurMember
+      
+      case pne@PNewExp(_) => issueError(pne, s"expected direct assignment")
```

Comment:
Is this the error I would get if I used new(...) for example inside a function or is there something that rejects that earlier? If this is the error I get, a better error message would be nice.